### PR TITLE
Selective enablement of tests

### PR DESCRIPTION
**Pull request by littlejawa**
- Add some code to the test script to selectively enable tests based on the version of cri-o being tested
- Enable several tests that we were skipping, for specific versions of cri-o
- Also enable a test fixed in kata - as we are testing against "main", there is no need to check the version here

### CODE REVIEW

**Comment by littlejawa** [comment]
I have a question that is not totally related to this PR, but more on the general git CI workflow.
When I make a change like this one, tests are being run, but as far as I can tell none of them is actually testing the thing that I modified (our test script is actually triggered only from cri-o PRs).
So it means that, even if the tests are successful, we have no way to tell whether I introduced a regression, right?

I'm not sure we want to run this script on every PR for the kata-containers/tests repository, but is there a way to check for regression at least when we modify something in that area?

**Comment by fidencio** [comment]
> I have a question that is not totally related to this PR, but more on the general git CI workflow. When I make a change like this one, tests are being run, but as far as I can tell none of them is actually testing the thing that I modified (our test script is actually triggered only from cri-o PRs). So it means that, even if the tests are successful, we have no way to tell whether I introduced a regression, right?

You're totally right.

> I'm not sure we want to run this script on every PR for the kata-containers/tests repository, but is there a way to check for regression at least when we modify something in that area?

One way to do that is introducing one specific job that would do something similar than what CRI-O does, running this PR against the `main` CRI-O branch.  In order to do so, we'd need to clone one specific CI, change its action call, and call it whenever doing such changes.

In case you need help to get this set up, please, do not hesitate on contacting me and I'd be happy to assist this process.

File: integration/cri-o/cri-o.sh [hunk 1]
```diff
@@ -74,6 +75,24 @@ do
 	sed -i '/'${testName}'/a skip \"'${skipCRIOTests[$testName]}'\"' "ctr_kata_integration_tests.bats"
 done
 
+# selectively skip tests depending on the version of cri-o we're testing with
+CRIO_VERSION=$(git status | head -n1 | awk '{print $NF}' | cut -d'-' -f 2)
+echo GOT CRIO VERSION $CRIO_VERSION
+if [ "$CRIO_VERSION" != "main" ]; then
+    if [ -z "$CRIO_VERSION" ]; then
+        echo "Unknown version of cri-o - skipping more tests"
+        CRIO_VERSION="1.21"
```
Comment (fidencio) [inline integration/cri-o/cri-o.sh:84]: I'd be more comfortable setting `CRIO_VERSION` to the latest release of the project.  In this case, 1.22.0 (see https://github.com/cri-o/cri-o/releases/tag/v1.22.0).

**Review by fidencio — COMMENTED** [review]
I like a lot the approach taken, @littlejawa!
There's one suggestion I made about the CRI-O version we should fallback to, and once that gets fixed and we ensure it doesn't break CRI-O CI this one will be good to go.

Thanks!

File: integration/cri-o/cri-o.sh (hunk 1 above)
Comment (littlejawa) [reply integration/cri-o/cri-o.sh:84]: Well, this is not the version we fall back to for running the test, this is the version we use to decide what test we enable or not.
1.21 is the first version we started testing with. And it is a version that we know have none of our fixes.

Using this number by default means we will skip all "unfixed" tests. If we're not sure what version of cri-o we're testing with, this is the safest. 

If we use "1.22" for instance, we will enable 3 more tests that are not fixed in 1.21, which means 3 failures if we're testing with a version of cri-o < 1.22

Comment (fidencio) [reply integration/cri-o/cri-o.sh:84]: Ack!  Thanks for the explanation.

**Review by fidencio — APPROVED** [review]

**Comment by jodh-intel** [comment]
@littlejawa - *Quis custodiet ipsos custodes?* :smile: We try to mitigate that risk, where possible, by having tests for the tests. For example, see:

- https://github.com/kata-containers/tests/blob/main/.ci/ci-fast-return.sh#L390..L599

You might be able to find further opportunities to add more "test tests" if you look around though ;)

**Review by jodh-intel — APPROVED** [review]
Thanks @littlejawa.

lgtm

**Pull request merged**